src/coverlet.msbuild.tasks/ReportWriter.cs

```diff
@@ -30,9 +30,10 @@ public string WriteReport()
 
       if (filename == string.Empty)
       {
-        // empty filename for instance only directory is passed to CoverletOutput c:\reportpath
-        // c:\reportpath\coverage.reportedextension
+        // empty filename for instance only directory is passed to CoverletOutput 'c:/reportpath/'
+        // note: use always '/' and not backslash => 'c:\reportpath\coverage.cobertura.xml'
         filename = $"coverage.{_coverletMultiTargetFrameworksCurrentTFM}{separatorPoint}{_reporter.Extension}";
+        _directory = Path.GetFullPath(_output);
```

Comment:
Sorry maybe a bit blind here but I don't see how this fixes #1610? And why is the full path needed again? In https://github.com/coverlet-coverage/coverlet/blob/master/src/coverlet.msbuild.tasks/CoverageResultTask.cs#L89 the directory path is already resolved and passed into the ctor of the `ReportWriter` as `directory` parameter.

Also I've seen that we have the same issue (#1610) with `coverlet.console`. But I don't think we need to tackle this in this PR here. I'll add it to the issue.

Reply:
`Path.GetFullPath(_output)` sanitizes the output path which is used for information message e.g. `/p:CoverletOutput="../Reports/coverlet.collector/"` will be

```
  Calculating coverage result...
  [coverlet] MergeWith: 'C:\GitHub\coverlet\artifacts\Reports\coverlet.core\coverage.json'.
   Generating report 'C:\GitHub\coverlet\test\Reports\coverlet.collector\coverage.net8.0.json'
   Generating report 'C:\GitHub\coverlet\test\Reports\coverlet.collector\coverage.net8.0.opencover.xml'
   
   and not 
   
  Calculating coverage result...
  [coverlet] MergeWith: 'C:\GitHub\coverlet\artifacts\Reports\coverlet.core\coverage.json'.
   Generating report '../Reports/coverlet.collector/coverage.net8.0.json'
   Generating report '../Reports/coverlet.collector/coverage.net8.0.opencover.xml'
   
```

Reply:
OK now I see what you want to do with this. But wouldn't it make more sense to sanitize the path somewhere later and not in the `(filename == string.Empty)` branch?
The paths with file name can also be relative and are currently also displayed like this in the information message.

![grafik](https://github.com/coverlet-coverage/coverlet/assets/19378678/1e889dd7-fc94-4f15-aafd-7bb6500172be)

I would probably do this right before the result gets written to the file.
```csharp
string report = Path.Combine(Path.GetFullPath(_directory), filename);
_fileSystem.WriteAllText(report, _reporter.Report(_result, _sourceRootTranslator));
```

Reply:
That is OK. I limited it to folder case only and kept the unrealistic  `Reporters.Msbuild_ReportWriter` tests. Removed the mix-up tests for folder and file name because the **default** for `_directory` is the current working directory of the test.